cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/sparksql/RowIterator.java

```diff
@@ -58,6 +65,55 @@ protected RowIterator(CellIterator it,
         this.builder = newBuilder(decorator);
     }
 
+    /**
+     * @param it              CellIterator cell iterate instance to iterate over each cell in the data.
+     * @param stats           stats instance to emit KPIs
+     * @param requiredColumns optional column filter to select a subset of columns, if null then all columns are returned.
+     * @return a basic RowIterator that transforms Object[] to a Map keyed on column name
+     */
+    public static RowIterator<Map<String, Object>> rowMapIterator(CellIterator it, Stats stats, @Nullable String[] requiredColumns)
+    {
+        final CqlTable cqlTable = it.cqlTable();
+        final boolean hasProjectedValueColumns = it.hasProjectedValueColumns();
+        return new RowIterator<>(it, stats, requiredColumns, Function.identity())
+        {
+            @Override
+            public PartialRowBuilder<Map<String, Object>> newPartialBuilder()
+            {
+                Objects.requireNonNull(requiredColumns, "requiredColumns must be non-null to use PartialRowBuilder");
+                Map<String, Integer> columnIndex = IntStream.range(0, requiredColumns.length)
+                                                            .boxed()
+                                                            .collect(Collectors.toMap(i -> requiredColumns[i], Function.identity()));
+
+                return new PartialRowBuilder<>(
+                requiredColumns, it.cqlTable(), hasProjectedValueColumns,
+                (valueArray) -> {
+                    Map<String, Object> row = new HashMap<>(requiredColumns.length);
+                    for (String field : requiredColumns)
+                    {
+                        row.put(field, valueArray[columnIndex.get(field)]);
+                    }
```

Comment:
The `columnIndex` map is probably not needed if you populate the row this way. 

```suggestion
                    for (int i = 0; i < requiredColumns.length; i++)
                    {
                        String field = requiredColumns[i];
                        row.put(field, valueArray[i]);
                    }
```